src/ioctls/vcpu.rs

```diff
@@ -19,6 +19,12 @@ use vmm_sys_util::ioctl::{ioctl, ioctl_with_mut_ref, ioctl_with_ref};
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 use vmm_sys_util::ioctl::{ioctl_with_mut_ptr, ioctl_with_ptr, ioctl_with_val};
 
+pub trait ValidRegisterType: Sized {}
```

Comment:
Does this need to be `unsafe`? The users are declaring their type meets all the properties required for being used as a register and using it with `get_one_reg` does not violate Rust's invariants. The compiler itself cannot check this.

Reply:
Marking this as unsafe will follow the same principle that we already apply to vm-memory: https://github.com/rust-vmm/vm-memory/blob/3ca7388bea433b05632ced19c82711f4d7d88a18/src/bytes.rs#L23.

Reply:
Good to hear.